includes/class-amp-theme-support.php

```diff
@@ -492,10 +470,32 @@ public static function start_output_buffering() {
 	 * Finish output buffering.
 	 *
 	 * @todo Do this in shutdown instead of output buffering callback?
+	 * @global int $content_width
 	 * @param string $output Buffered output.
 	 * @return string Finalized output.
 	 */
 	public static function finish_output_buffering( $output ) {
+		global $content_width;
+
+		$dom  = AMP_DOM_Utils::get_dom( $output );
+		$args = array(
+			'content_max_width' => ! empty( $content_width ) ? $content_width : AMP_Post_Template::CONTENT_MAX_WIDTH, // Back-compat.
+		);
+
+		/*
+		 * @todo The sanitize method needs to be updated to sanitize the entire HTML element and not just the BODY.
+		 * This will require updating mandatory_parent_blacklist in amphtml-update.py to include elements that appear in the HEAD.
+		 * This will ensure that the scripts and styles that plugins output via wp_head() will be sanitized as well. However,
+		 * since the the old paired mode is sending content from the *body* we'll need to be able to filter out the elements
+		 * from outside the body from being part of the whitelist sanitizer when it runs when theme support is not present,
+		 * as otherwise elements from the HEAD could get added to the BODY.
+		 */
+		list( $sanitized_inner_body, $scripts, $styles ) = AMP_Content_Sanitizer::sanitize( $dom, self::$sanitizer_classes, $args );
+
+		self::$amp_scripts = array_merge( self::$amp_scripts, $scripts );
+		self::$amp_styles  = array_merge( self::$amp_styles, $styles );
+
+		$output = preg_replace( '#(<body.*?>)(.+)(</body>)#si', '$1' . $sanitized_inner_body . '$3', $output );
```

Comment:
@westonruter here we're replacing the body with the sanitised body. This is because it's returning `AMP_DOM_Utils::get_content_from_dom( $dom )` from the sanitize method. Wouldn't this be easier if the method returned `$dom->saveHTML()` if the content passed is a complete document? This would stop the need to replace the body after it was sanitized.

Reply:
Yes, potentially. The `sanitize` method is better called `sanitize_content`. Better handling of sanitizing entire documents should be done.

Reply:
@DavidCramer how about this: https://github.com/Automattic/amp-wp/pull/888/commits/876c22fab1d70f9990ae42a52e1cc1bfd96c91d9

Reply:
@westonruter Thats nice, makes way for a full doc later on. I see that you made a comment about the `head` needs updating to `mandatory_parent_blacklist` so that makes sense now.